### PR TITLE
Added support to generate token of project

### PR DESCRIPTION
**Pull request by JoseAntonioTorre**
Now with endpoint /api/token/<project_id> you will get a {"token": "tokenofproject"} that you can use in GET "/api/project" with header "Authenthication: token <token>".

Hope it helps!

Issue: #412

### CODE REVIEW

**Comment by JoseAntonioTorre** [comment]
> Now with endpoint /api/token/<project_id> you will get a {"token": "tokenofproject"} that you can use in GET "/api/project" with header "Authenthication: token ".
> 
> Hope it helps!
> 
> Issue: #412

@almet I have branched from branch fix-control-public-creation instead of master. How can I fix it?

File: ihatemoney/api.py [hunk 1]
```diff
@@ -186,8 +186,21 @@ def delete(self, project, bill_id):
         return "OK", 200
 
 
+class TokenHandler(Resource):
+    method_decorators = [need_auth]
+
+    def get(self, project):
+        print(project)
```
Comment (almet) [inline ihatemoney/api.py]: It seems that this print shouldn't be here :-)

File: ihatemoney/api.py [hunk 2]
```diff
@@ -186,8 +186,21 @@ def delete(self, project, bill_id):
...
+        if not project:
+            return "Not Found", 404
+        token = project.generate_token()
+        print(token)
```
Comment (almet) [inline ihatemoney/api.py]: Extraneous print statement.

File: ihatemoney/api.py [hunk 3]
```diff
@@ -186,8 +186,21 @@ def delete(self, project, bill_id):
...
+        return {"token": token}, 200
+
+
 restful_api.add_resource(ProjectsHandler, "/projects")
 restful_api.add_resource(ProjectHandler, "/projects/<string:project_id>")
+restful_api.add_resource(TokenHandler, "/token/<string:project_id>")
```
Comment (almet) [inline ihatemoney/api.py]: Should we put this at `/projects/:id/token` instead?

File: ihatemoney/tests/tests.py [hunk 4]
```diff
@@ -1317,6 +1357,29 @@ def test_project(self):
...
+
+        decoded_resp = json.loads(resp.data.decode("utf-8"))
+
+        # Access with token
+        resp = self.client.get(
+            "/api/token/raclette",
+            headers={"Authorization": "Basic %s" % decoded_resp["token"]},
+        )
+
+        self.assertEqual(200, resp.status_code)
```
Comment (almet) [inline ihatemoney/tests/tests.py]: It seems to me that we should also test that the token can be used to login.

File: ihatemoney/api.py (hunk 1 above)
Comment (JoseAntonioTorre) [reply ihatemoney/api.py]: You are right!

File: ihatemoney/api.py (hunk 2 above)
Comment (JoseAntonioTorre) [reply ihatemoney/api.py]: you are right, only for testing

**Comment by JoseAntonioTorre** [comment]
> Here is some feedback:
> 
> * It seems that you also have code from your last pull request into this very one.
> * Some stuff I put in inline comments.
> 
> Thanks!

Thanks for your feedback it is very useful :) 
I will try to change it and I will come back.

I don't know how to modify your fist advice "it seems that you also have code from your last pull request into this very one". I created the branch from a feature-request branch instead of master one. How can I change it?

**Comment by almet** [comment]
> I don't know how to modify your fist advice "it seems that you also have code from your last pull request into this very one". I created the branch from a feature-request branch instead of master one. How can I change it?

I believe the simplest way is to re-create a branch from master (`git checkout master && git pull origin master && git checkout -b mynewbranch`) and then cherry-picking the commits you want from this branch (using `git cherry-pick`).

File: ihatemoney/tests/tests.py (hunk 4 above)
Comment (almet) [reply ihatemoney/tests/tests.py]: It should have been, but my review was too weak :(

**Pull request closed without merging**